inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/scatter_update/scatter_nd_update_kernel_ref.cpp

```diff
@@ -0,0 +1,234 @@
+    output_index_str.append("const uint z_remain = f_remain % ").append(std::to_string(Z_LEN)).append(";");
+    output_index_str.append("const uint w_remain = z_remain % ").append(std::to_string(W_LEN)).append(";");
+    output_index_str.append("const uint y_remain = w_remain % ").append(std::to_string(Y_LEN)).append(";");
+
+    output_index_str.append("const uint b = dst_idx / ").append(std::to_string(B_LEN)).append(";");
+    output_index_str.append("const uint f = b_remain / ").append(std::to_string(F_LEN)).append(";");
+    output_index_str.append("const uint z = f_remain / ").append(std::to_string(Z_LEN)).append(";");
+    output_index_str.append("const uint w = z_remain / ").append(std::to_string(W_LEN)).append(";");
+    output_index_str.append("const uint y = w_remain / ").append(std::to_string(Y_LEN)).append(";");
+    output_index_str.append("const uint x = y_remain;");
+
+    return output_index_str;
+}
+
+
+JitConstants ScatterNDUpdateKernelRef::GetJitConstants(const scatter_nd_update_params& params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+
+    if (!params.fused_ops.empty()) {
+        jit.AddConstant(MakeJitConstant("OUTPUT_INDEX_SECOND_KERNEL", GetOutputIndex(params)));
+
+        FusedOpsConfiguration conf1 = { "_FIRST_KERNEL", GetDefaultOrder(params.output.GetDims().size()), "val", params.inputs[0].GetDType() };
+        FusedOpsConfiguration conf2 = { "_SECOND_KERNEL", GetDefaultOrder(params.output.GetDims().size()), "val", params.inputs[0].GetDType() };
+        jit.Merge(MakeFusedOpsJitConstants(params, {conf1, conf2}));
+    }
+
+    return jit;
+}
+
+bool ScatterNDUpdateKernelRef::Validate(const Params& p, const optional_params& o) const {
+    if (p.GetType() != KernelType:: SCATTER_ND_UPDATE || o.GetType() != KernelType::SCATTER_ND_UPDATE) {
+        return false;
+    }
+
+    const scatter_nd_update_params& params = static_cast<const scatter_nd_update_params&>(p);
+
+    for (auto& fused_op : params.fused_ops) {
+        if (!IsFusedPrimitiveSupported(fused_op))
+            return false;
+    }
+
+    return true;
+}
+
+static std::string GetInputBlockND(const scatter_nd_update_params& params)
+{
+    const auto& input = params.inputs[0];
+    auto input_dims = input.LogicalDims();
+    std::reverse(input_dims.begin(), input_dims.end());
+    while (!input_dims.empty() && input_dims.back() == 1)
+    {
+        input_dims.pop_back();
+    }
+    const int rank = (int)input_dims.size();
+    std::vector<size_t> block_nd(rank + 1);
+    block_nd[rank] = 1;
+    for (int idx = (rank - 1); idx >= 0; idx--)
+    {
+        block_nd[idx] = input_dims[idx] * block_nd[idx + 1];
+    }
+
+    std::stringstream s;
+    for (int i = 0; i < (rank + 1); i++)
+    {
+        if (i < rank)
+        {
+            s << block_nd[i] << ",";
+        }
+        else
+        {
+            s << block_nd[i];
+        }
+    }
+    auto str_result = s.str();
+    return str_result;
+}
```

Comment:
is this trying to provide pitches for each input dim? it can be handled by INPUT0_X_PITCH, etc..